normandy/recipes/api/views.py

```diff
@@ -155,6 +144,101 @@ def disable(self, request, pk=None):
         recipe.disable()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @detail_route(methods=['GET'])
+    def approval_requests(self, request, pk=None):
+        recipe = self.get_object()
+        serializer = ApprovalRequestSerializer(recipe.approval_requests, many=True)
+        return Response(serializer.data)
+
+
+class ApprovalRequestViewSet(viewsets.ModelViewSet):
+    queryset = ApprovalRequest.objects.all()
+    serializer_class = ApprovalRequestSerializer
+    filter_fields = ('active',)
+    permission_classes = [
+        permissions.DjangoModelPermissionsOrAnonReadOnly,
+        AdminEnabledOrReadOnly,
+    ]
+
+    @transaction.atomic()
```

Comment:
There's gotta be a way to use a mixin or class decorator to auto-decorate viewset methods that modify things.